craft_parts/utils/os_utils.py

```diff
@@ -302,3 +302,22 @@ def version_codename(self) -> str:
             return _ID_TO_UBUNTU_CODENAME[self._os_release["VERSION_ID"]]
 
         raise errors.OsReleaseCodenameError()
+
+
+def process_run(command: List[str], log_func: Callable[[str], None], **kwargs) -> None:
+    """Run a command and handle its output."""
+    with subprocess.Popen(
+        command,
+        stdout=subprocess.PIPE,
+        stderr=subprocess.STDOUT,
+        universal_newlines=True,
+        **kwargs,
+    ) as proc:
+        if not proc.stdout:
+            return
+        for line in iter(proc.stdout.readline, ""):
+            log_func(line.strip())
```

Comment:
Why not log here in debug (without receiving the level from outside) and just use this function in the other modules, instead of defining two other intermediate functions?

I mean, everything will be simpler in how I propose it, why have it more complex? which is the benefit?

Reply:
Maybe we should prefix each line with something, so visually we understand that it's the output of the command? I used `:: ` in other codes and it's super clear.

Reply:
If call debug directly in `os_utils`, everything will be logged as originating from the `utils.os_util` package, not from the package where the code is actually running (`executor.step_handler`, `packages.deb`). This is temporary code anway, the final implementation should drop `process_run()` entirely and allow us to set stdout, stderr directly from `subprocess.run()`.

Reply:
Will add a prefix.